packer/rpc/communicator.go

```diff
@@ -97,7 +112,9 @@ func (c *communicator) Start(cmd *packer.RemoteCmd) (err error) {
 
 		var finished CommandFinished
 		decoder := gob.NewDecoder(conn)
-		if err := decoder.Decode(&finished); err != nil {
+		err = decoder.Decode(&finished)
+		wg.Wait()
```

Comment:
That makes sense to me, though I'm wondering if (specifically) a decoder error means we can't read the error response anyway. I can't be sure without exercising the branch conditions.